### PR TITLE
[FIX] website_sale: prevent from clicking Checkout too early (cart)

### LINKED ISSUE #29759: [12] Cart quantity is not updated on Process Checkout
Impacted versions: 12

Steps to reproduce: 

- Add Item to Cart
- Go to Cart
- Update Quantity
- Process Checkout

screenshot

Current behavior: Qty is not updated
sscreenshot

Expected behavior: Qty should be updated

Video/Screenshot link (optional):

### PR DESCRIPTION
**Pull request by seb-odoo**
The goal is to make sure the RPC to update quantity finishes before we start to
render the cart.

Indeed before this commit, the RPC would only be triggered on focus out. When
clicking on "Checkout" directly after typing a quantity manually, the RPC would
be made just before going to the checkout page, and it could happen that
the checkout page didn't display the new quantity because the RPC wasn't
finished yet.

With this change, we trigger the RPC as soon as possible, and we disable the
Checkout button until the RPC has finished.

closes #29759

### CODE REVIEW

**Comment by seb-odoo** [comment]
Good for me, this can be reviewed.

**Comment by rdeodoo** [comment]
@JKE-be 

> But on the other hand, this is not just another fast-click issue but a real use case that is not working (anyone willing to change quantity before buying will simply type new quantity and click directly on the Buy Now button without focusing out, then the new quantity will be ignored on next step, resulting in the cart summary before paying to be wrong (even if the good amount and qty will be send to the payment provider)).

User can't change quantity and then click on buy, that is a real use case honnestly.
That's not a traceback but it will just make the user to not understand.. "I typed 10 and then clicked on Buy Now but I still see 1 Quantity and 700$ instead of 7.000$".
I know that refreshing the page will render the correct amount and than going on a payment provider will send correct amount/qty but still.. User won't click on pay if that's not the quantity he typed, and having to refresh or going back on checkout flow is not a correct solution.

File: addons/website_sale/static/src/js/website_sale.js [hunk 1]
```diff
@@ -143,16 +155,44 @@ odoo.define('website_sale.website_sale', function (require) {
           var product_ids = [parseInt($input.data('product-id'),10)];
           clickwatch(function(){
             $dom_optional.each(function(){
+
+          // Prevent clicking on the Checkout button while the RPC is not done
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website_sale/static/src/js/website_sale.js]: Good catch! The indent on that file is not helping...

**Comment by qsm-odoo** [comment]
I agree it is a problem but it should be like 5-10 lines of code, not 45 ?
"on checkout click, wait for rpc before leaving", no need to disable stuff or anything to me, especially in stable versions.

**Comment by sswapnesh** [comment]
@JKE-be  Not even as `Wishlist` for future versions.

**Comment by vava-odoo** [comment]
It does not seem to be solved. The linked issue https://github.com/odoo/odoo/issues/29759 has been reopened.

**Pull request closed without merging**